source/code/modules/iceshard_pipelines/private/pipeline_ui/ip_ui_oven_elements.cxx

```diff
@@ -0,0 +1,398 @@
+                bool const action_parse_result = parse_action(
+                    action_value,
+                    button_info->action_on_click
+                );
+
+                ICE_ASSERT(
+                    action_parse_result != false,
+                    "Failed to parse action value '{}'",
+                    ice::String{ attribute->value(), attribute->value_size() }
+                );
+            }
+
+            attribute = attribute->next_attribute();
+        }
+
+        if (auto const* xml_node = xml_element->first_node("text"))
+        {
+            button_info->text = std::u8string_view{ reinterpret_cast<ice::c8utf const*>(xml_node->value()), xml_node->value_size() };
+        }
+
+        info.type_data = button_info;
+        info.type = ice::ui::ElementType::Button;
+    }
+
+
+    void compile_element_type(
+        ice::Allocator& alloc,
+        rapidxml_ns::xml_node<char> const* xml_element,
+        ice::RawElement& info
+    ) noexcept
+    {
+        info.type = ElementType::Page;
+        info.type_data = nullptr;
+
+        if (strcmp(xml_element->local_name(), "container") == 0)
+        {
+            rapidxml_ns::xml_attribute<char> const* const attrib = xml_element->first_attribute("type");
+            if (attrib != nullptr && strcmp(attrib->value(), "vertical") == 0)
+            {
+                info.type = ElementType::VListBox;
+            }
+        }
+        else if (strcmp(xml_element->local_name(), "button") == 0)
+        {
+            parse_element_type_button(xml_element, alloc, info);
+        }
+    }
+
+    void compile_element_attribs(
+        ice::Allocator& alloc,
+        rapidxml_ns::xml_node<char> const* element,
+        ice::RawElement& info
+    ) noexcept
+    {
+        using ice::ui::ElementFlags;
+        compile_element_type(alloc, element, info);
+
+        if (auto const* attr_size = element->first_attribute("size"); attr_size != nullptr)
+        {
+            parse_element_size(
+                attr_size->value(),
+                attr_size->value() + attr_size->value_size(),
+                info.size_flags,
+                info.size
+            );
+        }
+        else if (info.type == ElementType::Page)
+        {
+            info.size_flags = ElementFlags::Size_StretchWidth | ElementFlags::Size_StretchHeight;
+        }
+        else if (info.type == ElementType::VListBox)
+        {
+            info.size_flags = ElementFlags::Size_AutoWidth | ElementFlags::Size_AutoHeight;
+        }
+
+        if (auto const* attr_pos = element->first_attribute("position"); attr_pos != nullptr)
+        {
+            parse_element_pos(
+                attr_pos->value(),
+                attr_pos->value() + attr_pos->value_size(),
+                info.position_flags,
+                info.position
+            );
+        }
+
+        //if (auto const* attr_anchor = xml_element->first_attribute("anchor"); attr_anchor != nullptr)
```

Comment:
Might need to change this one into a follow up issue.

Reply:
For now we removed this code, as we might not need it for a longer time. If that's the case we would anyway need to re-visit most of the code.